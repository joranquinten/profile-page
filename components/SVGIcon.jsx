import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';

import atSVG from '../static/svg/at.svg';
import buildingSVG from '../static/svg/building.svg';
import cakeSVG from '../static/svg/cake.svg';
import coffeeSVG from '../static/svg/coffee.svg';
import githubSVG from '../static/svg/github.svg';
import guySVG from '../static/svg/guy.svg';
import linkedinSVG from '../static/svg/linkedin.svg';
import mediumSVG from '../static/svg/medium.svg';
import paperPlaneSVG from '../static/svg/paper-plane.svg';
import phoneSVG from '../static/svg/phone.svg';
import twitterSVG from '../static/svg/twitter.svg';

const iconMap = {
  at: atSVG,
  building: buildingSVG,
  cake: cakeSVG,
  coffee: coffeeSVG,
  github: githubSVG,
  guy: guySVG,
  linkedin: linkedinSVG,
  medium: mediumSVG,
  paperPlane: paperPlaneSVG,
  phone: phoneSVG,
  twitter: twitterSVG,
};

const createIcon = icon => ({ __html: iconMap[icon] });

const StyledIcon = styled.span`
  vertical-align: middle;
  display: inline-block;
  width: 24px;
  height: 24px;
  margin-right: 14px;
  line-height: 0;
`;

const Icon = props => {
  const { icon } = props;
  return <StyledIcon dangerouslySetInnerHTML={createIcon(icon)} />; // eslint-disable-line react/no-danger
};

Icon.propTypes = {
  icon: PropTypes.oneOf(Object.keys(iconMap)).isRequired,
};

export default Icon;
