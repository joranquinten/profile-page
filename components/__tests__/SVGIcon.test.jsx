import { shallow } from 'enzyme';
import SVGIcon from '../SVGIcon';

describe('<SVGIcon />', () => {
  it('should render without crashing', () => {
    expect(shallow(<SVGIcon icon="twitter" />)).toMatchSnapshot();
  });
});
