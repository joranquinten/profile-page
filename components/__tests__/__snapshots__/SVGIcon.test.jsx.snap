// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SVGIcon /> should render without crashing 1`] = `
<SVGIcon__StyledIcon
  dangerouslySetInnerHTML={
    Object {
      "__html": "twitter.svg",
    }
  }
/>
`;
